Ensure address validation button is visible (admin)

ryanreeves-taxjar opened the pull request:
Address validation was accidentally firing when searching for customers
during the order creation process.  Ensuring the button is visible
ensures it was actually clicked.

ryanreeves-taxjar committed:
Ensure address validation button is visble (admin)

Address validation was accidentally firing when searching for customers
during the order creation process.  Ensuring the button is visible
ensures it was actually clicked.

fastdivision commented on view/adminhtml/web/js/suggested_addresses_admin.js:
```diff
@@ -124,7 +124,10 @@ define([
                 e.preventDefault();
-                self.validateAddress($(this));
+
+                if (!button.is(':hidden')) {
```
@ryanreeves-taxjar Can you opt for `button.is(':visible')` instead so we can drop the `!` and make it easier to read? Thank you!

ryanreeves-taxjar committed:
Add type="button" to address validation button

The type="button" attribute was missing from the address validation
button, which caused AV to accidentally run when searching on a form.

fastdivision reviewed (approved):
Much better solution 👍
